src/core/search/search.cc

```diff
@@ -183,6 +194,31 @@ struct BasicSearch {
     return UnifyResults(GetSubResults(node.tags, mapping), LogicOp::OR);
   }
 
+  // [KNN count @field vec]: Compute distance to all vectors from query center and sort, keep
+  // closest `count`
+  IndexResult Search(const AstKnnNode& knn, string_view active_field) {
+    DCHECK(active_field.empty());
+    auto sub_results = SearchGeneric(*knn.filter, active_field);
+
+    auto* vec_index = GetIndex<VectorIndex>(knn.field);
+
+    vector<pair<float, DocId>> distances;
+    distances.reserve(sub_results->size());
+    for (DocId matched_doc : *sub_results) {
+      float dist = VectorDistance(knn.vector, vec_index->Get(matched_doc));
+      distances.emplace_back(dist, matched_doc);
+    }
+
+    sort(distances.begin(), distances.end());
+
+    vector<DocId> out(knn.limit);
```

Comment:
Do we want `std::min(knn.limit, distances.size())`?

Reply:
Good catch, adding a test